### PR TITLE
LRDOCS-4636 custom sys settings user interface draft

### PR DESCRIPTION
**Pull request by rbohl**

### CODE REVIEW

File: develop/tutorials/articles/290-configuration/02-custom-system-settings-ui.markdown [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+                String[] mysymbols = ParamUtil.getParameterValues(request, "mysymbols");
+
+                params.put("symbols", mysymbols);
+
+                return params;
+            }
+        }
+
+The above example ...
```
Comment (rbohl) [inline develop/tutorials/articles/290-configuration/02-custom-system-settings-ui.markdown]: @jorgeferrer What does this example do?

File: develop/tutorials/articles/290-configuration/02-custom-system-settings-ui.markdown [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+    @Reference(
+        target ="(osgi.web.symbolicname=com.liferay.currency.converter.web)", 
+        unbind = "-")
+    private ServletContext _servletContext;
+
+The most important step is to write the `render` method.This example relies on
+the `JSPRenderer` service to delegate rendering to a JSP.
+
+<!--Should we provide some sample JSP markup?-->
```
Comment (rbohl) [inline develop/tutorials/articles/290-configuration/02-custom-system-settings-ui.markdown]: @jorgeferrer do you think some example JSP (perhaps creating the multi-page configuration form mentioned in the intro) would be helpful here?

File: develop/tutorials/articles/290-configuration/02-custom-system-settings-ui.markdown (hunk 1 above)
Comment (JorgeFerrer) [reply develop/tutorials/articles/290-configuration/02-custom-system-settings-ui.markdown]: It generates a custom rendering (HTML) for the form (in render()) and reads the information entered in the form (in getRequestParameters())

File: develop/tutorials/articles/290-configuration/02-custom-system-settings-ui.markdown (hunk 2 above)
Comment (JorgeFerrer) [reply develop/tutorials/articles/290-configuration/02-custom-system-settings-ui.markdown]: Yes, definitely.

I think Drew should be able to help with that.

Comment (rbohl) [reply develop/tutorials/articles/290-configuration/02-custom-system-settings-ui.markdown]: cc @drewbrokke, care to provide some sample JSP for the docs? Seeing a concrete example will help devs consider the possibilities of the feature.

Comment (drewbrokke) [reply develop/tutorials/articles/290-configuration/02-custom-system-settings-ui.markdown]: Hey @rbohl sorry about the delay.  We don't have any examples ready to go so I will try to get them to you during this sprint.

Comment (rbohl) [reply develop/tutorials/articles/290-configuration/02-custom-system-settings-ui.markdown]: @drewbrokke I have a memory of you giving me some example code recently, but I can't find it in any of the usual places. Was I mis-remembering? If so, _bump_.

File: develop/tutorials/articles/290-configuration/02-custom-system-settings-ui.markdown [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+    @Reference(
+        target ="(osgi.web.symbolicname=com.liferay.currency.converter.web)", 
+        unbind = "-")
+    private ServletContext _servletContext;
+
+The most important step is to write the `render` method.This example relies on
+the `JSPRenderer` service to delegate rendering to a JSP.
+
+It's beyond the scope of this tutorial to write the JSP markup. A separate
```
Comment (rbohl) [inline develop/tutorials/articles/290-configuration/02-custom-system-settings-ui.markdown]: @drewbrokke @nealrobbins I propose saying this, or some such thing. We should, of course, immediately follow up with the next tutorial. What do you guys think?

Comment (drewbrokke) [reply develop/tutorials/articles/290-configuration/02-custom-system-settings-ui.markdown]: That could work.  Essentially it's the same as writing jsps for an MVCRenderCommand, so there would be a lot of overlap.  However, the main caveat I'd note somewhere (not necessarily here) is that if they write an MVCActionCommand to go with their JSP, they must register it with the System Settings portlet.  I did this in my sample JSP, if you need a reference.

**Comment by rbohl** [comment]
@drewbrokke @JorgeFerrer @nealrobbins

I added a demonstration tutorial on providing a fully custom UI. I consider it very much a draft. Please let me know if there are wrong statements, poorly explained bits, etc.

File: develop/tutorials/articles/290-configuration/03-fully-custom-system-settings-ui.markdown [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+the `addTemplateKey` method. Then set the other necessary values that the JSP
+will need. In this case, set the title, the field label, and the redirect URL.
+Lastly, call `renderJSP` and pass in the `servletContext`, request, response,
+and the path to the JSP: 
+
+    @Override
+    public void render(HttpServletRequest request, HttpServletResponse response)
+        throws IOException {
+
+        Locale locale = LocaleThreadLocal.getThemeDisplayLocale();
```
Comment (drewbrokke) [inline develop/tutorials/articles/290-configuration/03-fully-custom-system-settings-ui.markdown]: This variable should be moved down to where it's used around line 197.

File: develop/tutorials/articles/290-configuration/03-fully-custom-system-settings-ui.markdown [hunk 5]
```diff
@@ -0,0 +1,399 @@
...
+                ddmTemplate.getTemplateKey(), ddmTemplate.getName(locale));
+        }
+
+        languageTemplateConfigurationDisplayContext.setRedirect(
+            _portal.getCurrentURL(request));
+
+        languageTemplateConfigurationDisplayContext.setTitle(getName(locale));
+
+        languageTemplateConfigurationDisplayContext.setFieldLabel(
+            "Language Selection Style");
```
Comment (drewbrokke) [inline develop/tutorials/articles/290-configuration/03-fully-custom-system-settings-ui.markdown]: Whoops, forgot to make a language key for this.   I'd probably avoid doing a literal translation in our official documentation ;)  Sorry about that.

File: develop/tutorials/articles/290-configuration/03-fully-custom-system-settings-ui.markdown [hunk 6]
```diff
@@ -0,0 +1,399 @@
...
+        unbind = "-"
+    )
+    private ServletContext _servletContext;
+
+Once the configuration screen is implemented, render the form. This example
+uses a JSP.
+
+## Implementing an `MVCActionCommand` [](id=implementing-an-mvcactioncommand)
+
+For more detailed information on MVC Action Commnds, reea
```
Comment (drewbrokke) [inline develop/tutorials/articles/290-configuration/03-fully-custom-system-settings-ui.markdown]: typo here at `reea`?

File: develop/tutorials/articles/290-configuration/03-fully-custom-system-settings-ui.markdown [hunk 7]
```diff
@@ -0,0 +1,399 @@
...
+
+    <%
+    LanguageTemplateConfigurationDisplayContext languageTemplateConfigurationDisplayContext = (LanguageTemplateConfigurationDisplayContext)request.getAttribute(LanguageTemplateConfigurationDisplayContext.class.getName());
+
+    String currentTemplateName = languageTemplateConfigurationDisplayContext.getCurrentTemplateName();
+    %>
+
+    <portlet:actionURL name="/site_navigation_language/update_language_template_configuration" var="editURL" />
+
+    <div class="sheet sheet-lb">
```
Comment (drewbrokke) [inline develop/tutorials/articles/290-configuration/03-fully-custom-system-settings-ui.markdown]: `sheet-lb` should probably be `sheet-lg` according to http://pat270.github.io/lexicon/content/sheet/#sheet-1

File: develop/tutorials/articles/290-configuration/03-fully-custom-system-settings-ui.markdown (hunk 6 above)
Comment (rbohl) [reply develop/tutorials/articles/290-configuration/03-fully-custom-system-settings-ui.markdown]: yeah, and _commnds_ :disappointed:

Comment (drewbrokke) [reply develop/tutorials/articles/290-configuration/03-fully-custom-system-settings-ui.markdown]: Oh, good catch again!

**Comment by rbohl** [comment]
@drewbrokke @JorgeFerrer is this one ready to go? If either of you wants to do more review on it, let me know.

**Comment by rbohl** [comment]
Superceded: https://github.com/rbohl/liferay-docs/pull/96

**Pull request closed without merging**